File: exercises/js210_small_problems/easy_1/ex7.js
// function isLeapYear(year) {
//   if (year % 4 === 0 ) {
//     if (year % 100 === 0) {
//       if (year % 400 === 0) return true;
//       return false;
//     }
//     return true;
//   } else {
//     return false;
//   }
// }

isLeapYear = (year) => (year % 400 === 0) || (year % 4 === 0 && year % 100 !== 0);

console.log(isLeapYear(2016));      // true
console.log(isLeapYear(2015));      // false
console.log(isLeapYear(2100));      // false
console.log(isLeapYear(2400));      // true
console.log(isLeapYear(240000));    // true
console.log(isLeapYear(240001));    // false
console.log(isLeapYear(2000));      // true
console.log(isLeapYear(1900));      // false
console.log(isLeapYear(1752));      // true
console.log(isLeapYear(1700));      // false
console.log(isLeapYear(1));         // false
console.log(isLeapYear(100));       // false
console.log(isLeapYear(400));       // true

File: exercises/rb101_119/medium2/ex7.rb
=begin

Write a method that returns the number of Friday the 13ths in the year given by an argument. You may assume that the year is greater than 1752.

input: integer
output: integer
rules:
  explicit:
    - input integer is year
    - output i number of Friday the 13ths in the year
    - the input is always greter than 1752

DATA
Date class

ALGORITHM
- requre the Date class
- define a method `friday_13th(year)`
- set a `date` variable asssigned to the first day of the input year
- set a counter variable `number` to keep count of the number of Friday the 13ths
- start a loop until the year in `date` is one greter than input year
  - if the day is Friday and the 13th
    - increment `number` by one
  - reassign `date` to the next  day
- return `number`

=end

require 'date'

def friday_13th(year)
  date = Date.new(year)
  fridays = 0

  until date.year == year + 1
    fridays += 1 if date.friday? && date.day == 13
    date = date.next_day
  end

  fridays
end

p friday_13th(2015) == 3
p friday_13th(1986) == 1
p friday_13th(2019) == 2
